bridges/modules/dispatch/src/lib.rs

```diff
@@ -599,9 +599,11 @@ mod tests {
 	fn should_fail_on_weight_mismatch() {
 		new_test_ext().execute_with(|| {
 			let id = [0; 4];
-			let mut message =
-				prepare_root_message(Call::System(<frame_system::Call<TestRuntime>>::remark(vec![1, 2, 3])));
+			let call = Call::System(<frame_system::Call<TestRuntime>>::remark(vec![1, 2, 3]));
+			let call_weight = call.get_dispatch_info().weight;
+			let mut message = prepare_root_message(call);
 			message.weight = 7;
+			assert!(call_weight != 7, "needed for test to be correct");
```

Comment:
Can you explain the comment here a little more?

Reply:
the test is testing for a weight mismatch, so we must not match the correct weight